### PR TITLE
Added `onContentValidateData` event.

### PR DESCRIPTION
**Pull request by BrainforgeUK**
Pull Request for Issue #19584.

### Summary of Changes

Added event onContentValidateData

### Testing Instructions

Create a field plugin which implements onContentValidateData function like this...
public function onContentValidateData($form, &$data) {
// Change the data in some way
}

### Expected result

The data get's changed.

### Actual result

The data get's changed.

### Documentation Changes Required

Add documentation for onContentValidateData event.
Add documentation for the existing, but undocumented and now deprecated, onUserBeforeDataValidation event.

### CODE REVIEW

**Comment by BrainforgeUK** [comment]
Also added com_install change commit for 'Improved extension update error message' by accident. This applies to Joomla issue #22762.
I am struggling with how to create 2 pull requests for separate issues.
-------------------------------------
Undone the mentioned com_install change.

File: libraries/src/MVC/Model/FormModel.php [hunk 1]
```diff
@@ -192,7 +192,18 @@ public function validate($form, $data, $group = null)
 		// Include the plugins for the delete events.
 		PluginHelper::importPlugin($this->events_map['validate']);
 
-		Factory::getApplication()->triggerEvent('onUserBeforeDataValidation', array($form, &$data));
+		$dispatcher = Factory::getContainer()->get('dispatcher');
+
+		if (!empty($dispatcher->getListeners('onUserBeforeDataValidation'))) {
+		    @trigger_error(
```
Comment (Quy) [inline libraries/src/MVC/Model/FormModel.php]:
```suggestion
			@trigger_error(
```

File: libraries/src/MVC/Model/FormModel.php [hunk 2]
```diff
@@ -192,7 +192,18 @@ public function validate($form, $data, $group = null)
...
+					'The `onUserBeforeDataValidation` event is deprecated and will be removed in 5.0, use the `onContentValidateData` event instead.',
+					E_USER_DEPRECATED
+				);
+
+		    Factory::getApplication()->triggerEvent('onUserBeforeDataValidation', array($form, &$data));
```
Comment (Quy) [inline libraries/src/MVC/Model/FormModel.php]:
```suggestion
			Factory::getApplication()->triggerEvent('onUserBeforeDataValidation', array($form, &$data));
```

**Comment by SharkyKZ** [comment]
>I am struggling with how to create 2 pull requests for separate issues.

Instead of making changes to 4.0-dev branch, create separate branches for each PR.

File: libraries/src/MVC/Model/FormModel.php [hunk 3]
```diff
@@ -192,7 +192,18 @@ public function validate($form, $data, $group = null)
 		// Include the plugins for the delete events.
 		PluginHelper::importPlugin($this->events_map['validate']);
 
-		Factory::getApplication()->triggerEvent('onUserBeforeDataValidation', array($form, &$data));
+		$dispatcher = Factory::getContainer()->get('dispatcher');
+
+		if (!empty($dispatcher->getListeners('onUserBeforeDataValidation'))) {
```
Comment (Quy) [inline libraries/src/MVC/Model/FormModel.php]:
```suggestion
		if (!empty($dispatcher->getListeners('onUserBeforeDataValidation')))
		{
```

File: libraries/src/MVC/Model/FormModel.php [hunk 4]
```diff
@@ -192,7 +192,18 @@ public function validate($form, $data, $group = null)
...
+			@trigger_error(
+					'The `onUserBeforeDataValidation` event is deprecated and will be removed in 5.0, use the `onContentValidateData` event instead.',
```
Comment (Quy) [inline libraries/src/MVC/Model/FormModel.php]: Reduce indent by one with lines 199-201.

Comment (BrainforgeUK) [reply libraries/src/MVC/Model/FormModel.php]: Changed - see libraries\src\Cache\CacheController.php line 125 for existing similar.

**Comment by BrainforgeUK** [comment]
Replaced by pull request #25103

**Pull request closed without merging**